File: src/test/java/test/LitecartLoginTestBase.java
package test;

import java.io.IOException;

import org.openqa.selenium.By;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import static org.openqa.selenium.support.ui.ExpectedConditions.titleIs;

public abstract class LitecartLoginTestBase extends BrowserTestBase {

    public LitecartLoginTestBase(WebBrowserCode browserCode){
        super(browserCode);
    }

    @BeforeClass
    public void openBrowser() {
        try {
            Runtime.getRuntime().exec(SettingsKt.getRunServerString());
        } catch (IOException e) {
            e.printStackTrace();
        }

        initDriver();
    }

    @Test
    public void checkLitecartLogin() {
        final String ADMIN = "admin";

        getDriver().get("http://localhost/litecart/admin/");
        getDriver().findElement(By.name("username")).sendKeys(ADMIN);
        getDriver().findElement(By.name("password")).sendKeys(ADMIN);
        getDriver().findElement(By.name("login")).click();
        getDriverWait().until(titleIs("My Store"));
    }

    @AfterClass
    public void closeBrowser() {
        cleanDriver();

        try {
            Runtime.getRuntime().exec(SettingsKt.getStopServerString());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/kotlin/test/LitecartLoginTestBase.kt
package test

import org.openqa.selenium.By
import org.openqa.selenium.support.ui.ExpectedConditions.titleIs
import org.testng.annotations.AfterClass
import org.testng.annotations.BeforeClass
import org.testng.annotations.Test
import java.io.IOException

abstract class LitecartLoginTestBase(browser: WebBrowserCode) : BrowserTestBase(browser) {
    @BeforeClass
    fun openBrowser() {
        try {
            Runtime.getRuntime().exec(getRunServerString())
        } catch (e: IOException) {
            e.printStackTrace()
        }

        initDriver()
    }

    @Test
    fun checkLitecartLogin() {
        val admin = "admin"

        driver.get("http://localhost/litecart/admin/")
        driver.findElement(By.name("username")).sendKeys(admin)
        driver.findElement(By.name("password")).sendKeys(admin)
        driver.findElement(By.name("login")).click()
        driverWait.until<Boolean>(titleIs("My Store"))
    }

    @AfterClass
    fun closeBrowser() {
        cleanDriver()

        try {
            Runtime.getRuntime().exec(getStopServerString())
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}